### PR TITLE
handling of exception added

### LINKED ISSUE #1527: Importer: Unhandled exception when "Reset based on current sheet"
[TestData.xlsx](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/files/7828854/TestData.xlsx)

- Open the attached file for import
- Select the second sheet "Mitrakou_1990_EGP"
- "Reset Mapping based on current sheet"

```
************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at OSPSuite.Presentation.Presenters.Importer.ColumnMappingPresenter.SetDataFormat(IDataFormat format)
   at OSPSuite.Presentation.Presenters.Importer.ImporterPresenter.onFormatChanged(Object sender, FormatChangedEventArgs e)
   at System.EventHandler`1.Invoke(Object sender, TEventArgs e)
   at OSPSuite.Presentation.Presenters.Importer.ImporterDataPresenter.SetDataFormat(IDataFormat format, IEnumerable`1 availableFormats)
   at OSPSuite.Presentation.Presenters.Importer.ImporterDataPresenter.GetFormatBasedOnCurrentSheet()
   at OSPSuite.Presentation.Presenters.Importer.ImporterPresenter.onResetMappingBasedOnCurrentSheet()
   at OSPSuite.Presentation.Presenters.Importer.ImporterPresenter.<>c__DisplayClass15_0.<.ctor>b__4(Object o, EventArgs e)
   at System.EventHandler.Invoke(Object sender, EventArgs e)
   at OSPSuite.Presentation.Presenters.Importer.ColumnMappingPresenter.ResetMappingBasedOnCurrentSheet()
   at OSPSuite.UI.Views.Importer.ColumnMappingView.onResetMappingCurrentSheetClick(Object sender, EventArgs eventArgs)
   at DevExpress.Utils.Menu.DXMenuItem.OnClick()
   at DevExpress.XtraBars.Utils.DXPopupXtraMenu.OnBarMenuClick(Object sender, ItemClickEventArgs e)
   at DevExpress.XtraBars.BarItem.OnClick(BarItemLink link)
   at DevExpress.XtraBars.BarButtonItem.OnClick(BarItemLink link)
   at DevExpress.XtraBars.BarItemLink.OnLinkClick()
   at DevExpress.XtraBars.BarButtonItemLink.OnLinkAction(BarLinkAction action, Object actionArgs)
   at DevExpress.XtraBars.ViewInfo.BarSelectionInfo.ClickLink(BarItemLink link)
   at DevExpress.XtraBars.ViewInfo.BarSelectionInfo.UnPressLink(BarItemLink link)
   at DevExpress.XtraBars.Controls.CustomLinksControl.OnMouseUp(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at DevExpress.XtraBars.Controls.CustomControl.WndProc(Message& msg)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

### PR DESCRIPTION
**Pull request by georgeDaskalakis**
closes #1527

### CODE REVIEW

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs [hunk 1]
```diff
@@ -260,7 +260,12 @@ public string GetActiveFilterCriteria()
 
       public void GetFormatBasedOnCurrentSheet()
       {
-         _dataSourceFile.AvailableFormats = _importer.CalculateFormat(_dataSourceFile, _columnInfos, _metaDataCategories, _currentSheetName).ToList();
+         var availableFormats  = _importer.CalculateFormat(_dataSourceFile, _columnInfos, _metaDataCategories, _currentSheetName).ToList();
+         
+         if(availableFormats.Count <1 )
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs:265]: if(!availableFormats.Any()) is typically what is used here. Also resharper should be used in this file

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs [hunk 2]
```diff
@@ -272,4 +273,26 @@ public void first_valid_sheet_is_selected()
...
+   public class When_calculating_format_based_on_invalid_sheet : concern_for_ImporterDataPresenter
+   {
+      protected override void Context()
+      {
+         base.Context();
+         sut.SetDataSource("test_file");
+      }
+
+      protected override void Because()
+      {
```
Comment (msevestre) [inline tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs:286]: Please use resharper in this file too. There is spaces extra everywhere it seems

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs [hunk 3]
```diff
@@ -272,4 +273,26 @@ public void first_valid_sheet_is_selected()
...
+
+         A.CallTo(() => _importer.CalculateFormat(A<IDataSourceFile>._ ,A<ColumnInfoCache>._, A<IReadOnlyList<MetaDataCategory>>._, A<string>._ )).Returns(new List<IDataFormat>());
+      }
+
+      [Observation]
+      public void action_should_not_proceed()
+      {
+         Assert.Throws<UnsupportedFormatException>(() => sut.GetFormatBasedOnCurrentSheet());
```
Comment (msevestre) [inline tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs:294]: Use The.Action(()=>xxx.) shoudThrow

**Review by msevestre — CHANGES_REQUESTED** [review]
See comments

**Comment by msevestre** [comment]
@georgeDaskalakis Some minor changs and also conflicts after all those PR merged. Can you fix them (don't forget to run resharper on all files, ImporterDataPresenter.cs and ImporterDataPresenterSpecs.cs)

**Comment by msevestre** [comment]
@georgeDaskalakis Can you take care of this

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs (hunk 1 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs:265]: done and format also fixed through Resharper

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs (hunk 2 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs:286]: done

File: tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs (hunk 3 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Presentation.Tests/Importer/Presenters/ImporterDataPresenterSpecs.cs:294]: done

**Comment by georgeDaskalakis** [comment]
@msevestre this should be ready. sorry for the delay, was not working the previous 2 days

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs [hunk 4]
```diff
@@ -83,15 +81,16 @@ public void ImportDataForConfirmation(string sheetName)
             Sheets.Add(sheetName, getSingleSheet(sheetName));
             sheets.Add(sheetName, getSingleSheet(sheetName));
          }
-         if (sheets.Count == 0) 
-            return;
 
-         OnImportSheets.Invoke(this, new ImportSheetsEventArgs { DataSourceFile = _dataSourceFile, Sheets = sheets, Filter = GetActiveFilterCriteria()});
+         if (sheets.Count == 0) return;
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs:85]: Why are all those return on one line now? Is your resharper config doing this? Can you use the one I sent you a while back? i can definitely sent it back to you if needed

File: src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs [hunk 5]
```diff
@@ -226,13 +221,11 @@ public void RemoveAllButThisTab(string tabName)
...
             DataSheet currentAlreadyLoaded = null;
-            if (Sheets.Keys.Contains(tabName))
-               currentAlreadyLoaded = Sheets[tabName];
+            if (Sheets.Keys.Contains(tabName)) currentAlreadyLoaded = Sheets[tabName];
 
             Sheets.Clear();
-            if (currentAlreadyLoaded != null)
-               Sheets.Add(tabName, currentAlreadyLoaded);
-            
+            if (currentAlreadyLoaded != null) Sheets.Add(tabName, currentAlreadyLoaded);
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs:227]: no this needs to be all undone here. We do not want statements on one line like this at all.
Please undo this commit

**Review by msevestre — CHANGES_REQUESTED** [review]
a bunch of ifs are now on one line. This is a no no and goes against our coding standards

**Comment by georgeDaskalakis** [comment]
@msevestre format corrected. I had to do it by hand, because ReSharper would not change the one line ifs and return when simply applying the format, but I ve been careful about it, so this PR should be ready.

**Pull request merged**